osu.Framework/Audio/Callbacks/DataStreamFileProcedures.cs

```diff
@@ -3,17 +3,16 @@
 
 using System;
 using System.IO;
-using System.Runtime.InteropServices;
+
+#nullable enable
 
 namespace osu.Framework.Audio.Callbacks
 {
     /// <summary>
     /// Implementation of <see cref="IFileProcedures"/> that supports reading from a <see cref="Stream"/>.
     /// </summary>
-    public class DataStreamFileProcedures : IFileProcedures
+    internal class DataStreamFileProcedures : IFileProcedures
```

Comment:
This is regressing osu!-side usage for checking too short audio files in editor:

<img width="1665" alt="CleanShot 2022-05-04 at 15 01 37@2x" src="https://user-images.githubusercontent.com/22781491/166677568-15edfbe5-9bb8-49ce-a1f5-1de8eff6a4c7.png">

```suggestion
    public class DataStreamFileProcedures : IFileProcedures
```

Reply:
I'd just revert the modifier change honestly.

Reply:
Should we add a null check then, probably just in constructor?

Reply:
`?? throw new ArgumentNullException` makes sense given that the game is not fully NRT-enabled.